flathub.json

```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
+}
```

Comment:
why restrict it to x86_64 ?

Reply:
Because it fails to build in aarch64.

```
dbus-run-session flatpak run org.flatpak.Builder -v --force-clean --sandbox --delete-build-dirs --user --install-deps-from=flathub --install-deps-from=flathub-beta --arch aarch64 --mirror-screenshots-url=https://dl.flathub.org/media --repo repo --extra-sources=/srv/buildbot/worker/build-aarch64-6/../downloads --default-branch test --subject 'Update io.github.mhogomchungu.sirikali.json (becf397c)' builddir io.github.mhogomchungu.sirikali.json --download-only
 in dir /srv/buildbot/worker/build-aarch64-6/build (timeout 3600 secs)
 watching logfiles {}
 argv: [b'dbus-run-session', b'flatpak', b'run', b'org.flatpak.Builder', b'-v', b'--force-clean', b'--sandbox', b'--delete-build-dirs', b'--user', b'--install-deps-from=flathub', b'--install-deps-from=flathub-beta', b'--arch', b'aarch64', b'--mirror-screenshots-url=https://dl.flathub.org/media', b'--repo', b'repo', b'--extra-sources=/srv/buildbot/worker/build-aarch64-6/../downloads', b'--default-branch', b'test', b'--subject', b'Update io.github.mhogomchungu.sirikali.json (becf397c)', b'builddir', b'io.github.mhogomchungu.sirikali.json', b'--download-only']
 using PTY: True

Note that the directories 

'/var/lib/flatpak/exports/share'
'/srv/buildbot/.local/share/flatpak/exports/share'

are not in the search path set by the XDG_DATA_DIRS environment variable, so
applications installed by Flatpak may not appear on your desktop until the
session is restarted.

dbus-daemon[1209071]: [session uid=996 pid=1209071] Activating service name='org.freedesktop.Flatpak' requested by ':1.0' (uid=996 pid=1209072 comm="flatpak run org.flatpak.Builder -v --force-clean -" label="unconfined")
dbus-daemon[1209071]: [session uid=996 pid=1209071] Successfully activated service 'org.freedesktop.Flatpak'
dbus-daemon[1209071]: [session uid=996 pid=1209071] Activating service name='org.freedesktop.portal.Documents' requested by ':1.0' (uid=996 pid=1209072 comm="flatpak run org.flatpak.Builder -v --force-clean -" label="unconfined")
dbus-daemon[1209071]: [session uid=996 pid=1209071] Activating service name='org.freedesktop.impl.portal.PermissionStore' requested by ':1.2' (uid=996 pid=1209083 comm="/usr/libexec/xdg-document-portal " label="unconfined")
dbus-daemon[1209071]: [session uid=996 pid=1209071] Successfully activated service 'org.freedesktop.impl.portal.PermissionStore'
dbus-daemon[1209071]: [session uid=996 pid=1209071] Successfully activated service 'org.freedesktop.portal.Documents'
fusermount3: failed to access mountpoint /run/user/996/doc: Permission denied
error: fuse init failed: Can't mount path /run/user/996/doc
F: Can't get document portal: GDBus.Error:org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying
dbus-daemon[1209071]: [session uid=996 pid=1209071] Activating service name='org.a11y.Bus' requested by ':1.0' (uid=996 pid=1209072 comm="flatpak run org.flatpak.Builder -v --force-clean -" label="unconfined")

** (process:1209096): [1;31mERROR[0m **: [34m19:14:08.915[0m: Cannot get the default GSettingsSchemaSource - is the gsettings-desktop-schemas package installed?
dbus-daemon[1209071]: [session uid=996 pid=1209071] Activated service 'org.a11y.Bus' failed: Process org.a11y.Bus received signal 5
F: Can't find a11y bus: GDBus.Error:org.freedesktop.DBus.Error.Spawn.ChildSignaled: Process org.a11y.Bus received signal 5
FB: Running: git config --get user.email
FB: Running: git config --get user.name
Dependency Sdk: org.kde.Sdk 6.7
FB: Running 'flatpak --user info org.kde.Sdk/aarch64/6.7 --show-commit' on host
FB: host_command_exited_cb 1209118 0

Updating org.kde.Sdk/aarch64/6.7
FB: Using Flatpak version 1.14.10
FB: Running 'flatpak --user update --subpath= -y --noninteractive org.kde.Sdk/aarch64/6.7' on host
Info: org.freedesktop.Sdk.Extension.rust-nightly is end-of-life, with reason: This extension should not be used for packaging purposes.

Nothing to do.
FB: host_command_exited_cb 1209124 0

Dependency Runtime: org.kde.Platform 6.7
FB: Running 'flatpak --user info org.kde.Platform/aarch64/6.7 --show-commit' on host
FB: host_command_exited_cb 1209129 0

Updating org.kde.Platform/aarch64/6.7
FB: Running 'flatpak --user update --subpath= -y --noninteractive org.kde.Platform/aarch64/6.7' on host

Nothing to do.
FB: host_command_exited_cb 1209132 0

FB: Running 'flatpak info --arch=aarch64 --show-commit org.kde.Sdk 6.7' on host
FB: host_command_exited_cb 1209138 0

FB: Running 'flatpak info --show-location --arch=aarch64 org.kde.Sdk 6.7' on host
FB: host_command_exited_cb 1209142 0

FB: Running 'flatpak info --arch=aarch64 --show-commit org.kde.Platform 6.7' on host
FB: host_command_exited_cb 1209145 0

Downloading sources
]2;flatpak-builder: Downloading libfuse]2;flatpak-builder: Downloading io.github.mhogomchungu.sirikaliDownloading https://github.com/mhogomchungu/sirikali/releases/download/1.6.0/sirikali-1.6.0.2.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 1575k  100 1575k    0     0  6779k      0 --:--:-- --:--:-- --:--:-- 6779k
program finished with exit code 0
elapsedTime=3.765950
dbus-run-session flatpak run org.flatpak.Builder -v --force-clean --sandbox --delete-build-dirs --user --install-deps-from=flathub --install-deps-from=flathub-beta --arch aarch64 --mirror-screenshots-url=https://dl.flathub.org/media --repo repo --extra-sources=/srv/buildbot/worker/build-aarch64-6/../downloads --default-branch test --subject 'Update io.github.mhogomchungu.sirikali.json (becf397c)' builddir io.github.mhogomchungu.sirikali.json --disable-download
 in dir /srv/buildbot/worker/build-aarch64-6/build (timeout 3600 secs)
 watching logfiles {}
 argv: [b'dbus-run-session', b'flatpak', b'run', b'org.flatpak.Builder', b'-v', b'--force-clean', b'--sandbox', b'--delete-build-dirs', b'--user', b'--install-deps-from=flathub', b'--install-deps-from=flathub-beta', b'--arch', b'aarch64', b'--mirror-screenshots-url=https://dl.flathub.org/media', b'--repo', b'repo', b'--extra-sources=/srv/buildbot/worker/build-aarch64-6/../downloads', b'--default-branch', b'test', b'--subject', b'Update io.github.mhogomchungu.sirikali.json (becf397c)', b'builddir', b'io.github.mhogomchungu.sirikali.json', b'--disable-download']
 using PTY: True

Note that the directories 

'/var/lib/flatpak/exports/share'
'/srv/buildbot/.local/share/flatpak/exports/share'

are not in the search path set by the XDG_DATA_DIRS environment variable, so
applications installed by Flatpak may not appear on your desktop until the
session is restarted.

dbus-daemon[1209158]: [session uid=996 pid=1209158] Activating service name='org.freedesktop.Flatpak' requested by ':1.0' (uid=996 pid=1209159 comm="flatpak run org.flatpak.Builder -v --force-clean -" label="unconfined")
dbus-daemon[1209158]: [session uid=996 pid=1209158] Successfully activated service 'org.freedesktop.Flatpak'
dbus-daemon[1209158]: [session uid=996 pid=1209158] Activating service name='org.freedesktop.portal.Documents' requested by ':1.0' (uid=996 pid=1209159 comm="flatpak run org.flatpak.Builder -v --force-clean -" label="unconfined")
dbus-daemon[1209158]: [session uid=996 pid=1209158] Activating service name='org.freedesktop.impl.portal.PermissionStore' requested by ':1.2' (uid=996 pid=1209170 comm="/usr/libexec/xdg-document-portal " label="unconfined")
dbus-daemon[1209158]: [session uid=996 pid=1209158] Successfully activated service 'org.freedesktop.impl.portal.PermissionStore'
dbus-daemon[1209158]: [session uid=996 pid=1209158] Successfully activated service 'org.freedesktop.portal.Documents'
fusermount3: failed to access mountpoint /run/user/996/doc: Permission denied
error: fuse init failed: Can't mount path /run/user/996/doc
F: Can't get document portal: GDBus.Error:org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying
dbus-daemon[1209158]: [session uid=996 pid=1209158] Activating service name='org.a11y.Bus' requested by ':1.0' (uid=996 pid=1209159 comm="flatpak run org.flatpak.Builder -v --force-clean -" label="unconfined")

** (process:1209183): [1;31mERROR[0m **: [34m19:14:15.240[0m: Cannot get the default GSettingsSchemaSource - is the gsettings-desktop-schemas package installed?
dbus-daemon[1209158]: [session uid=996 pid=1209158] Activated service 'org.a11y.Bus' failed: Process org.a11y.Bus received signal 5
F: Can't find a11y bus: GDBus.Error:org.freedesktop.DBus.Error.Spawn.ChildSignaled: Process org.a11y.Bus received signal 5
FB: Running: git config --get user.email
FB: Running: git config --get user.name
Dependency Sdk: org.kde.Sdk 6.7
FB: Running 'flatpak --user info org.kde.Sdk/aarch64/6.7 --show-commit' on host
FB: host_command_exited_cb 1209205 0

Updating org.kde.Sdk/aarch64/6.7
FB: Using Flatpak version 1.14.10
FB: Running 'flatpak --user update --subpath= -y --noninteractive org.kde.Sdk/aarch64/6.7' on host
Info: org.freedesktop.Sdk.Extension.rust-nightly is end-of-life, with reason: This extension should not be used for packaging purposes.

Nothing to do.
FB: host_command_exited_cb 1209210 0

Dependency Runtime: org.kde.Platform 6.7
FB: Running 'flatpak --user info org.kde.Platform/aarch64/6.7 --show-commit' on host
FB: host_command_exited_cb 1209215 0

Updating org.kde.Platform/aarch64/6.7
FB: Running 'flatpak --user update --subpath= -y --noninteractive org.kde.Platform/aarch64/6.7' on host

Nothing to do.
FB: host_command_exited_cb 1209218 0

FB: Running 'flatpak info --arch=aarch64 --show-commit org.kde.Sdk 6.7' on host
FB: host_command_exited_cb 1209224 0

FB: Running 'flatpak info --show-location --arch=aarch64 org.kde.Sdk 6.7' on host
FB: host_command_exited_cb 1209228 0

FB: Running 'flatpak info --arch=aarch64 --show-commit org.kde.Platform 6.7' on host
FB: host_command_exited_cb 1209231 0

Initializing build dir
FB: Running: flatpak build-init --arch=aarch64 /srv/buildbot/worker/build-aarch64-6/build/builddir io.github.mhogomchungu.sirikali org.kde.Sdk org.kde.Platform 6.7
Committing stage init to cache
Starting build of io.github.mhogomchungu.sirikali
FB: Mounting read-only fs: rofiles-fuse /srv/buildbot/worker/build-aarch64-6/build/builddir /srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/rofiles/rofiles-HNQSBq
========================================================================
Building module libfuse in /srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/build/libfuse-1
========================================================================
]2;flatpak-builder: Building libfuseFB: Running: tar xf /srv/buildbot/worker/downloads/downloads/d0e69d5d608cc22ff4843791ad097f554dd32540ddc9bed7638cc6fea7c1b4b5/fuse-2.9.9.tar.gz --no-same-owner --strip-components=1 -z
FB: Running 'flatpak build --die-with-parent --env=FLATPAK_BUILDER_BUILDDIR=/run/build/libfuse --nofilesystem=host:reset --filesystem=/srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/build/libfuse-1 --bind-mount=/run/build/libfuse=/srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/build/libfuse-1 --build-dir=/run/build/libfuse --bind-mount=/run/ccache=/srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/ccache --unshare=network --env=SOURCE_DATE_EPOCH=1727032178 '--env=CFLAGS=-O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer ' '--env=CXXFLAGS=-O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer ' '--env=LDFLAGS=-L/app/lib -Wl,-z,relro,-z,now -Wl,--as-needed ' --env=CCACHE_DIR=/run/ccache/disabled --env=PATH=/app/bin:/usr/bin --env=LD_LIBRARY_PATH=/app/lib --env=PKG_CONFIG_PATH=/app/lib/pkgconfig:/app/share/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig --env=FLATPAK_BUILDER_N_JOBS=80 /srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/rofiles/rofiles-HNQSBq ./configure --prefix=/app --disable-util' on host
checking build system type... aarch64-unknown-linux-gnu
checking host system type... aarch64-unknown-linux-gnu
checking target system type... aarch64-unknown-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking how to print strings... printf
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /usr/aarch64-unknown-linux-gnu/bin/ld
checking if the linker (/usr/aarch64-unknown-linux-gnu/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert aarch64-unknown-linux-gnu file names to aarch64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert aarch64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/aarch64-unknown-linux-gnu/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/aarch64-unknown-linux-gnu/bin/ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking for fork... yes
checking for setxattr... yes
checking for fdatasync... yes
checking for splice... yes
checking for vmsplice... yes
checking for utimensat... yes
checking for posix_fallocate... yes
checking for struct stat.st_atim... yes
checking for struct stat.st_atimespec... no
checking for library containing dlopen... none required
checking for library containing clock_gettime... none required
checking for ld used by gcc... /usr/aarch64-unknown-linux-gnu/bin/ld
checking if the linker (/usr/aarch64-unknown-linux-gnu/bin/ld) is GNU ld... yes
checking for shared library run path origin... done
checking for iconv... yes
checking for working iconv... yes
checking for iconv declaration... 
         extern size_t iconv (iconv_t cd, char * *inbuf, size_t *inbytesleft, char * *outbuf, size_t *outbytesleft);
configure: MOUNT_FUSE_PATH env var not set, using default /sbin
configure: UDEV_RULES_PATH env var not set, using default /etc/udev/rules.d
configure: INIT_D_PATH env var not set, using default /etc/init.d
checking if umount supports --fake --no-canonicalize... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating fuse.pc
config.status: creating Makefile
config.status: creating lib/Makefile
config.status: creating util/Makefile
config.status: creating example/Makefile
config.status: creating include/Makefile
config.status: creating doc/Makefile
config.status: creating include/config.h
config.status: executing depfiles commands
config.status: executing libtool commands
FB: host_command_exited_cb 1209248 0

]2;flatpak-builder: Installing libfuseFB: Running 'flatpak build --die-with-parent --env=FLATPAK_BUILDER_BUILDDIR=/run/build/libfuse --nofilesystem=host:reset --filesystem=/srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/build/libfuse-1 --bind-mount=/run/build/libfuse=/srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/build/libfuse-1 --build-dir=/run/build/libfuse --bind-mount=/run/ccache=/srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/ccache --unshare=network --env=SOURCE_DATE_EPOCH=1727032178 '--env=CFLAGS=-O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer ' '--env=CXXFLAGS=-O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer ' '--env=LDFLAGS=-L/app/lib -Wl,-z,relro,-z,now -Wl,--as-needed ' --env=CCACHE_DIR=/run/ccache/disabled --env=PATH=/app/bin:/usr/bin --env=LD_LIBRARY_PATH=/app/lib --env=PKG_CONFIG_PATH=/app/lib/pkgconfig:/app/share/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig --env=FLATPAK_BUILDER_N_JOBS=80 /srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/rofiles/rofiles-HNQSBq make -j80 -l160' on host
Making all in include
make[1]: Entering directory '/run/build/libfuse/include'
make  all-am
make[2]: Entering directory '/run/build/libfuse/include'
make[2]: Leaving directory '/run/build/libfuse/include'
make[1]: Leaving directory '/run/build/libfuse/include'
Making all in lib
make[1]: Entering directory '/run/build/libfuse/lib'
  CC       fuse.lo
  CC       fuse_loop.lo
  CC       fuse_kern_chan.lo
  CC       fuse_loop_mt.lo
  CC       fuse_lowlevel.lo
  CC       fuse_mt.lo
  CC       fuse_session.lo
  CC       fuse_opt.lo
  CC       fuse_signals.lo
  CC       buffer.lo
  CC       cuse_lowlevel.lo
  CC       helper.lo
  CC       mount.lo
  CC       mount_util.lo
  CC       ulockmgr.lo
  CC       modules/iconv.lo
  CC       modules/subdir.lo
mount_util.c: In function ‘mtab_needs_update’:
mount_util.c:62:25: warning: ignoring return value of ‘setreuid’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   62 |                         setreuid(0, -1);
      |                         ^~~~~~~~~~~~~~~
mount_util.c:67:25: warning: ignoring return value of ‘setreuid’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   67 |                         setreuid(ruid, -1);
      |                         ^~~~~~~~~~~~~~~~~~
mount_util.c: In function ‘add_mount’:
mount_util.c:102:17: warning: ignoring return value of ‘setuid’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  102 |                 setuid(geteuid());
      |                 ^~~~~~~~~~~~~~~~~
mount_util.c: In function ‘exec_umount’:
mount_util.c:155:17: warning: ignoring return value of ‘setuid’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  155 |                 setuid(geteuid());
      |                 ^~~~~~~~~~~~~~~~~
In file included from fuse_loop_mt.c:11:
../include/fuse_kernel.h:93:15: error: conflicting types for ‘int64_t’; have ‘long long int’
   93 | #define __s64 int64_t
      |               ^~~~~~~
In file included from /usr/include/stdint.h:34,
                 from /usr/lib/gcc/aarch64-unknown-linux-gnu/13.2.0/include/stdint.h:9,
                 from ../include/fuse_common.h:19,
                 from ../include/fuse_lowlevel.h:26,
                 from fuse_loop_mt.c:9:
/usr/include/aarch64-linux-gnu/bits/stdint-intn.h:27:19: note: previous declaration of ‘int64_t’ with type ‘int64_t’ {aka ‘long int’}
   27 | typedef __int64_t int64_t;
      |                   ^~~~~~~
mount_util.c: In function ‘remove_mount’:
../include/fuse_kernel.h:92:15: error: conflicting types for ‘uint64_t’; have ‘long long unsigned int’
   92 | #define __u64 uint64_t
      |               ^~~~~~~~
mount_util.c:221:17: warning: ignoring return value of ‘setuid’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  221 |                 setuid(geteuid());
      |                 ^~~~~~~~~~~~~~~~~
In file included from /usr/include/stdint.h:37:
/usr/include/aarch64-linux-gnu/bits/stdint-uintn.h:27:20: note: previous declaration of ‘uint64_t’ with type ‘uint64_t’ {aka ‘long unsigned int’}
   27 | typedef __uint64_t uint64_t;
      |                    ^~~~~~~~
helper.c: In function ‘fuse_daemonize’:
helper.c:203:25: warning: ignoring return value of ‘read’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  203 |                         read(waiter[0], &completed, sizeof(completed));
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
helper.c:212:24: warning: ignoring return value of ‘chdir’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  212 |                 (void) chdir("/");
      |                        ^~~~~~~~~~
helper.c:225:17: warning: ignoring return value of ‘write’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  225 |                 write(waiter[1], &completed, sizeof(completed));
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[1]: *** [Makefile:518: fuse_loop_mt.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/run/build/libfuse/lib'
make: *** [Makefile:450: all-recursive] Error 1
FB: host_command_exited_cb 1211354 512

]2;flatpak-builder: Cleanup libfuseError: module libfuse: Child process exited with code 2
FB: Unmounting read-only fs: fusermount -uz /srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/rofiles/rofiles-HNQSBq
program finished with exit code 1
elapsedTime=57.937112
```

Reply:
Add these patches https://github.com/flathub/org.flatpak.Builder.BaseApp/blob/5c36bb2b3b216ca54af2f0e194bb5beee4b3ee14/org.flatpak.Builder.BaseApp.json#L420-L427

Also did you set up the fusermount wrapper script ?

Reply:
No need for wrapper scripts because flatpak is treated as a platform and its specifics are handled in the code.

Example of where a command to unmount a filesystem is constructed.

https://github.com/mhogomchungu/sirikali/blob/master/src/engines.cpp#L1323-L1335

Example of where securefs is constructing a string to unlock a volume.

https://github.com/mhogomchungu/sirikali/blob/1a86b10d554c49533b68a433a55b1072230861a7/src/engines/securefs.cpp#L88-L103